orderer/common/multichannel/registrar.go

```diff
@@ -651,7 +670,19 @@ func (r *Registrar) createAsMember(ledgerRes *ledgerResources, configBlock *cb.B
 		r.bccsp,
 	)
 	if err != nil {
-		return nil, types.ChannelInfo{}, errors.Wrap(err, "error creating chain support")
+		err = errors.Wrap(err, "error creating chain support")
+
+		if err2 := r.ledgerFactory.Remove(channelID); err2 != nil {
+			// Stack ledger removal error on top of chain support creation error
+			err = errors.WithStack(err2)
+		}
+
+		if err2 := r.removeJoinBlock(channelID); err2 != nil {
+			// Stack file repo error on top of ledger removal + chain support creation error
+			err = errors.WithStack(err2)
+		}
```

Comment:
you can put this in a defer that checks the return `err` right after the join-block is saved

Reply:
Done